File: topicos/dynamic-programming/group-1/arithmetic_series/arithmetic_series.c
#include <stdio.h>

long long arithmetic_series(long long n, long long a1, long long 2n) {
    long long serie = a1 + int a, n;

    if ((n & 1ULL) == 0) {
        return (long long)(n >> 1) * serie;
    } else {
        return (long long)n * (serie >> 1)
    }
}

File: topicos/dynamic-programming/group-1/arithmetic_series/arithmetic_series.rs
pub fn arithmetic_series(n: u64, a1: i64) -> i64 {
    let serie = a1 + an;

    if n&1 == 0 {
        (n>>1) as i64 * serie 
    } else {
        n as i64 * (serie>>1)
    }
}